### PR TITLE
Press compress

### PR DESCRIPTION
**Pull request by jlark**
## Updates:
- Adding gzip compression from RMC to API #14 
## Notes:
- Kraken will look for headers `Content-Encoding` = `gzip` and will attempt to decompress body if this exists.
  ## Changes:
- Due to problems decompressing string values format of bulk updates have have changed
### Old format:

`http://url:port/input/bulk.json?api_key=<key>&site_id=<site_id>`
- with POST body:
  `data= <JSON_DATA>`
  `time=<TIMESTAMP>`
### New format:

`http://url:port/input/bulk.json?api_key=<key>,site_id=<site_id>&time=<TIMESTAMP>`
- with POST body:
  `<JSON_DATA>`

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -105,19 +107,30 @@ def post():
...
 def bulk():
-    data = request.form['data']
+
+
+    # Check if compressesed
+    if request.headers.get('Content-Encoding', None) == 'gzip':
+        data= zlib.decompress(request.get_data())
+        logging.debug("decompressed %s "%data)
+    else:
+        data = request.get_data()
```
Comment (bumi) [inline api.py]: I think we should move this in a `before_request` method that it applies to all the endpoints.

**Comment by bumi** [comment]
oh nice, I like the new format better anyway... If possible I think it would be great to only have POST requests with JSON bodies.

File: api.py (hunk 1 above)
Comment (jlark) [inline api.py]: Good idea.

Comment (jlark) [inline api.py]: Done

**Comment by bumi** [comment]
great. can you delete that file with name `1` ? 
then merge?

**Comment by jlark** [comment]
well that was embarasing :sake:

**Comment by bumi** [comment]
want to make it mergable or merge it?

**Pull request merged**